egs/librispeech/s5/local/chain/run_tdnn_6z.sh

```diff
@@ -0,0 +1,208 @@
+set -e
+
+# configs for 'chain'
+affix=
+stage=11
+train_stage=-10
+speed_perturb=true
+get_egs_stage=-10
+dir=exp/chain/tdnn_6z  # Note: _sp will get added to this if $speed_perturb == true.
+decode_iter=
+
+# TDNN options
+# this script uses the new tdnn config generator so it needs a final 0 to reflect that the final layer input has no splicing
+# smoothing options
+pool_window=
+pool_type='none'
+pool_lpfilter_width=
+self_repair_scale=0.00001
+# training options
+num_epochs=4
+initial_effective_lrate=0.001
+final_effective_lrate=0.0001
+leftmost_questions_truncate=-1
+max_param_change=2.0
+final_layer_normalize_target=0.5
+num_jobs_initial=3
+num_jobs_final=16
+minibatch_size=128
+relu_dim=576
+frames_per_eg=150
+remove_egs=false
+common_egs_dir=
+xent_regularize=0.1
+
+
+
+# End configuration section.
+echo "$0 $@"  # Print the command line for logging
+
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+# The iVector-extraction and feature-dumping parts are the same as the standard
+# nnet3 setup, and you can skip them by setting "--stage 8" if you have already
+# run those things.
+
+suffix=
+if [ "$speed_perturb" == "true" ]; then
+  suffix=_sp
+fi
+
+dir=${dir}${affix:+_$affix}
+dir=${dir}$suffix
+train_set=train_960$suffix
+ali_dir=exp/tri6b$suffix
+treedir=exp/chain/tri5_2y_tree$suffix
+lang=data/lang_chain_2y
+
+
+# if we are using the speed-perturbed data we need to generate
+# alignments for it.
+local/nnet3/run_ivector_common.sh --stage $stage \
```

Comment:
Yes for xent recipe, no for chain recipe.

On Mon, Apr 4, 2016 at 3:57 PM, Yiming Wang <notifications@github.com>
wrote:

> In egs/librispeech/s5/local/chain/run_tdnn_6z.sh
> <https://github.com/kaldi-asr/kaldi/pull/645#discussion_r58437498>:
>
> > +suffix=
> > +if [ "$speed_perturb" == "true" ]; then
> > +  suffix=_sp
> > +fi
> > +
> > +dir=${dir}${affix:+_$affix}
> > +dir=${dir}$suffix
> > +train_set=train_960$suffix
> > +ali_dir=exp/tri6b$suffix
> > +treedir=exp/chain/tri5_2y_tree$suffix
> > +lang=data/lang_chain_2y
> > +
> > +
> > +# if we are using the speed-perturbed data we need to generate
> > +# alignments for it.
> > +local/nnet3/run_ivector_common.sh --stage $stage \
>
> Do we need to set generate_aligment to true if speed_perturb == true?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/645/files/9ea0ff3e250cec5ed8f22b1a837c17791398eaf6#r58437498>
>